ignition/src/IgnitionRobot.cpp

```diff
@@ -149,6 +157,85 @@ JointEntity IgnitionRobot::Impl::getJointEntity(const JointName& jointName)
+        // The hierarchy must be: model -> link -> collision
+        auto linkEntity = ecm->ParentEntity(collisionEntity);
+        auto modelEntity = ecm->ParentEntity(linkEntity);
+        return getEntityName(modelEntity);
+    };
+
+    ecm->Each<ignition::gazebo::components::Collision,
+              ignition::gazebo::components::ContactSensorData,
+              ignition::gazebo::components::ParentEntity>(
+        [&](const ignition::gazebo::Entity& /*collisionEntity*/,
+            ignition::gazebo::components::Collision* /*collisionComponent*/,
+            ignition::gazebo::components::ContactSensorData* contactDataComponent,
+            ignition::gazebo::components::ParentEntity* linkEntityComponent) -> bool {
+            // Get the model entity of the link that contains the collision
+            auto modelEntity = ecm->ParentEntity(linkEntityComponent->Data());
+
+            // Keep only collisions of links that belong to the handled model
+            if (modelEntity != model.Entity()) {
+                return true;
+            }
+
+            // Get the name of the link associated to this collision entity
+            std::string linkName =
+                ecm->Component<ignition::gazebo::components::Name>(linkEntityComponent->Data())
+                    ->Data();
+
+            // Extract the contacts data structure
+            ignition::msgs::Contacts contactsData = contactDataComponent->Data();
+
+            assert(buffers.links.contacts.find(linkName) != buffers.links.contacts.end());
+            buffers.links.contacts[linkName].clear();
+            buffers.links.contacts[linkName].reserve(contactsData.contact_size());
+
+            // Process each contact
+            for (int i = 0; i < contactsData.contact_size(); ++i) {
+                // Extract the contact object
+                const ignition::msgs::Contact& contactData = contactsData.contact(i);
+
+                std::string scopedBodyA =
+                    getParentModelNameFromCollisionEntity(contactData.collision1().id())
+                    + "::" + getEntityName(contactData.collision1().id());
+
+                std::string scopedBodyB =
+                    getParentModelNameFromCollisionEntity(contactData.collision2().id())
+                    + "::" + getEntityName(contactData.collision2().id());
+
+                // Extract the contact data
+                ContactData contact;
+                contact.bodyA = scopedBodyA;
+                contact.bodyB = scopedBodyB;
+                contact.position[0] = contactData.position(i).x();
+                contact.position[1] = contactData.position(i).y();
+                contact.position[2] = contactData.position(i).z();
+
+                // TODO: fill the normal
+                // TODO: fill the contacty depth
+                // TODO: fill the wrench magnitude
+                assert(contactData.depth_size() == 0);
```

Comment:
This methods do not seem to be defined. Are you sure that this code compiles in debug mode?

Reply:
Sorry for the naming confusion. `contactData` is an object of `ignition::msgs::Contact` (line 206) and it does have those methods.

Reply:
Ahhha, you are right.